File: main_module/src/main/java/com/phone/main_module/login/request/LoginRequest.java
package com.phone.main_module.login.request;

import com.phone.common_library.common.ConstantUrl;

import java.util.Map;

import io.reactivex.Observable;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.http.Body;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;

public interface LoginRequest {
	
	//    @FormUrlEncoded
	@GET(ConstantUrl.GET_VERIFICATION_CODE_URL)
	Observable<ResponseBody> getAuthCode(@QueryMap Map<String, String> bodyParams);
	
	@FormUrlEncoded
	@POST(ConstantUrl.LOGIN_WITH_AUTH_CODE_URL)
	Observable<ResponseBody> getLoginWithAuthCodeData(@FieldMap Map<String, String> bodyParams);
	
	@FormUrlEncoded
	@POST(ConstantUrl.REGISTER_URL)
	Observable<ResponseBody> getRegisterData(@FieldMap Map<String, String> bodyParams);
	
	//    @Multipart
	//    @POST("upload")
	//    Observable<JSONObject> addShopResult(@PartMap Map<String, RequestBody> fileMap,
	//                                              @PartMap Map<String, List<RequestBody>> filesMap);
	
	@POST(ConstantUrl.ADD_SHOP_URL)
	Observable<ResponseBody> addShopResult(@Body RequestBody requestBody);
	
}

File: library_login/src/main/java/com/phone/library_login/request/LoginRequest.kt
package com.phone.library_login.request

import com.phone.library_common.common.ConstantUrl
import io.reactivex.Observable
import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.http.*

interface LoginRequest {

    //    @FormUrlEncoded
    @GET(ConstantUrl.GET_VERIFICATION_CODE_URL)
    fun getAuthCode(@QueryMap bodyParams: Map<String, String>): Observable<ResponseBody>

    @FormUrlEncoded
    @POST(ConstantUrl.LOGIN_WITH_AUTH_CODE_URL)
    fun getLoginWithAuthCodeData(@FieldMap bodyParams: Map<String, String>): Observable<ResponseBody>

    @FormUrlEncoded
    @POST(ConstantUrl.REGISTER_URL)
    fun getRegisterData(@FieldMap bodyParams: Map<String, String>): Observable<ResponseBody>

    //    @Multipart
    //    @POST("upload")
    //    Observable<JSONObject> addShopResult(@PartMap Map<String, RequestBody> fileMap,
    //                                              @PartMap Map<String, List<RequestBody>> filesMap);

    //    @Multipart
    //    @POST("upload")
    //    Observable<JSONObject> addShopResult(@PartMap Map<String, RequestBody> fileMap,
    //                                              @PartMap Map<String, List<RequestBody>> filesMap);

    @POST(ConstantUrl.ADD_SHOP_URL)
    fun addShopResult(@Body requestBody: RequestBody): Observable<ResponseBody>

}